sync-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -248,53 +249,52 @@ public Boolean call() {
         }
 
         List<DBParameter> parameters = new ArrayList<DBParameter>();
-        if (arrayCount == 0) {
-            // The are no arrays in the values we are indexing. We just need to index the fields
-            // in the index. _id and _rev are special fields in that they don't appear in the
-            // body, so they need special-casing to get the values.
-
-            List<String> initialIncludedFields = new ArrayList<String>();
-            initialIncludedFields.add("_id");
-            initialIncludedFields.add("_rev");
-            List<Object> initialArgs = new ArrayList<Object>();
-            initialArgs.add(rev.getId());
-            initialArgs.add(rev.getRevision());
-            DBParameter parameter = populateDBParameter(fieldNames,
-                                                        initialIncludedFields,
-                                                        initialArgs,
-                                                        indexName,
-                                                        rev);
-            if (parameter == null) {
-                return null;
-            }
-            parameters.add(parameter);
-        } else if (arrayFieldName != null) {
-            // We know the value is an array, we found this out in the check above
+
+        if (arrayCount == 1) {
+            // Value is an array
             List<Object> arrayFieldValues;
-            arrayFieldValues = (ArrayList) ValueExtractor.extractValueForFieldName(arrayFieldName,
-                                                                                   rev.getBody());
-            for (Object value: arrayFieldValues) {
-                List<String> initialIncludedFields = new ArrayList<String>();
-                initialIncludedFields.add("_id");
-                initialIncludedFields.add("_rev");
-                initialIncludedFields.add(arrayFieldName);
-                List<Object> initialArgs = new ArrayList<Object>();
-                initialArgs.add(rev.getId());
-                initialArgs.add(rev.getRevision());
-                initialArgs.add(value);
-                DBParameter parameter;
-                parameter = populateDBParameter(fieldNames,
-                                                initialIncludedFields,
-                                                initialArgs,
-                                                indexName,
-                                                rev);
-                if (parameter == null) {
-                    return null;
+            arrayFieldValues = (List) ValueExtractor.extractValueForFieldName(arrayFieldName,
+                                                                              rev.getBody());
+            if (arrayFieldValues != null && arrayFieldValues.size() > 0) {
```

Comment:
So, I think that if arrayFieldValues is null, then something terrible must have happened, right? It seems impossible. 

But having that check here makes the code confusing, no?

Perhaps my following suggestion could be a bit of overkill, but could you create a Map of Lists, like `mapOfArrayFieldValues` and populate it during the initial loop through `fieldNames`. Then check the number of objects in the map instead of having `arrayCount` and `arrayFieldName` would be the only key in `mapOfArrayFieldValues`. 

But the code does work as written. I was just thrown a bit by the check for null. Or maybe I'm wrong at it can by null here?